x-pack/plugins/fleet/server/services/spaces/enable_space_awareness.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+
+import { LEGACY_AGENT_POLICY_SAVED_OBJECT_TYPE } from '../../../common';
+import {
+  LEGACY_PACKAGE_POLICY_SAVED_OBJECT_TYPE,
+  PACKAGE_POLICY_SAVED_OBJECT_TYPE,
+} from '../../../common/constants';
+
+import { appContextService } from '..';
+import { AGENT_POLICY_SAVED_OBJECT_TYPE } from '../../constants';
+import { getSettingsOrUndefined, saveSettings } from '../settings';
+import { FleetError } from '../../errors';
+
+const PENDING_TIMEOUT = 60 * 60 * 1000;
```

Comment:
IMO this concept of a timeout is not entirely necessary. We have the ability to map to: `pending | error | complete` and our code should be robust to ending in one of those states. Then the timeout becomes less useful.

Reply:
@jloleysens I think the issue we could have if is the process running the migration before completing we will end in a state that is not recoverable unless we have that timeout concept.